platform/darwin/src/MGLGeometryOfflineRegion.mm

```diff
@@ -0,0 +1,120 @@
+#import "MGLGeometryOfflineRegion.h"
+
+#if !TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR
+    #import <Cocoa/Cocoa.h>
+#else
+    #import <UIKit/UIKit.h>
+#endif
+
+#import "MGLOfflineRegion_Private.h"
+#import "MGLGeometryOfflineRegion_Private.h"
+#import "MGLFeature_Private.h"
+#import "MGLShape_Private.h"
+#import "MGLStyle.h"
+
+@interface MGLGeometryOfflineRegion () <MGLOfflineRegion_Private, MGLGeometryOfflineRegion_Private>
+
+@end
+
+@implementation MGLGeometryOfflineRegion {
+    NSURL *_styleURL;
+}
+
+@synthesize styleURL = _styleURL;
```

Comment:
You should be able to replace the `@synthesize` and the instance var declaration with a single `@property`

Reply:
Wouldn't accept just the property, so left this for now.

Reply:
👍
 
This ought to work, so I'd be interested in why it might not be for you:
```
@interface MGLShapeOfflineRegion () <MGLOfflineRegion_Private, MGLShapeOfflineRegion_Private>
@property (nonatomic, copy) NSURL *styleURL;
@end
```

Reply:
It probably has to do with the fact that the property’s getter is declared in a protocol rather than a concrete class, which would prevent autosynthesis. If that’s the case, then explicit synthesis is fine.